src/platform/i386/chal_pgtbl.h

```diff
@@ -0,0 +1,184 @@
+/* These code below are for x86 specifically */
+typedef enum {
+	X86_PRESENT    = 1,
```

Comment:
See my previous comments on 1. namespacing for this, and 2. it seems that we should override the general page-table flags for each architecture, so that the translation between generic and architecture flags is as easy/fast as it can be.  I'd rather not add complexity here except where it is necessary.  If you don't quite get my intention here, we can talk about it tomorrow.